test.js

```diff
@@ -66,31 +96,80 @@ test('cpr timeout', function (t) {
     t.equal(aircraft.callsign, '')
 
     t.end()
+  })
+})
+
+test('cpr timeout with reference time', function (t) {
+  const store = new AircraftStore()
 
-    Date.now = origNow
+  // if more than 10 seconds between messages, lat/lng decoding will be
+  // skipped, so lets simulate a long delay between messages
+  const options = {
+    refTime: 4815162342,
+    timeBetweenMessages: 11000
+  }
+
+  populateStore(store, options, function () {
+    const aircrafts = store.getAircrafts(options.refTime)
+    t.equal(aircrafts.length, 1)
+
+    const aircraft = aircrafts[0]
+    t.equal(aircraft.icao, 7867807)
+    t.equal(aircraft.count, msgs.length)
+    t.ok(aircraft.seen >= options.refTime)
+    t.equal(aircraft.altitude, 31000)
+    t.equal(aircraft.unit, 0)
+    t.equal(aircraft.speed, 506.66359648192605)
+    t.equal(aircraft.heading, 65.76194226683805)
+    t.equal(aircraft.lat, 0)
+    t.equal(aircraft.lng, 0)
+    t.equal(aircraft.callsign, '')
+
+    t.end()
   })
 })
 
-test('store timeout', function (t) {
+test('store timeout with system time', function (t) {
   const store = new AircraftStore({timeout: 50})
+  const start = Date.now()
 
-  populateStore(store, function () {
-    const aircrafts = store.getAircrafts()
-    t.equal(aircrafts.length, 1)
-    setTimeout(function () {
-      const aircrafts = store.getAircrafts()
-      t.equal(aircrafts.length, 0)
-      t.end()
-    }, 51)
+  const options = {
+    refTime: null,
+    timeBetweenMessages: 0
+  }
+
+  populateStore(store, options, function () {
+    t.equal(store.getAircrafts().length, 1)
+    t.equal(store.getAircrafts(start + 51).length, 0)
+    t.end()
   })
 })
 
-function populateStore (store, cb, _decoder, _index) {
-  if (!_decoder) _decoder = new Decoder()
-  if (!_index) _index = 0
-  if (msgs.length === _index) return cb()
-  setTimeout(function () {
-    store.addMessage(_decoder.parse(msgs[_index]))
-    populateStore(store, cb, _decoder, ++_index)
-  }, 1)
+test('store timeout with reference time', function (t) {
+  const store = new AircraftStore({timeout: 50})
+
+  const options = {
+    refTime: 4815162342,
+    timeBetweenMessages: 0
+  }
+
+  populateStore(store, options, function () {
+    t.equal(store.getAircrafts(options.refTime).length, 1)
+    t.equal(store.getAircrafts(options.refTime + 51).length, 0)
+    t.end()
+  })
+})
+
+function populateStore (store, options, cb) {
+  options.decoder = options.decoder || new Decoder()
+  options.refTime = options.refTime || Date.now()
+
+  for (let i = 0; i < msgs.length; i++) {
+    store.addMessage(
+      options.decoder.parse(msgs[i]),
+      options.refTime + options.timeBetweenMessages * i
```

Comment:
All tests end up passing in a custom time as the 2nd argument, so we really don't get to test the built in fallback to system time. Consider calling `store.addMessage` inside a `setTimeout` if no `options.refTime` is provided instead of falling back to `Date.now()`.